stackslib/src/chainstate/stacks/boot/mod.rs

```diff
@@ -1126,6 +1133,38 @@ impl StacksChainState {
             x => x,
         }
     }
+
+    /// Get the aggregate public key for a given reward cycle from pox 4
+    pub fn get_aggregate_public_key_pox_4(
+        &mut self,
+        sortdb: &SortitionDB,
+        block_id: &StacksBlockId,
+        reward_cycle: u64,
+    ) -> Result<Option<Point>, Error> {
+        if !self.is_pox_active(sortdb, block_id, reward_cycle as u128, POX_4_NAME)? {
+            debug!(
+                "PoX was voted disabled in block {} (reward cycle {})",
+                block_id, reward_cycle
+            );
+            return Ok(None);
+        }
+
+        let aggregate_public_key = self
+            .eval_boot_code_read_only(
+                sortdb,
+                block_id,
+                POX_4_NAME,
+                &format!("(get-aggregate-pubilc-key u{})", reward_cycle),
```

Comment:
Fix typo in call

Reply:
Is there a `get-aggregate-public-key` function in `pox-4`?  If not, can you add it?

Also, I don't think this is ever going to be optional in PoX-4.  Instead, this method should just return `Result<Point, Error>`.

Reply:
This could be optional for a given reward cycle as signers may not have set it yet no? In theory they should have set it BEFORE the start of their reward cycle, but they may fail to do so before the start.

Reply:
> In theory they should have set it BEFORE the start of their reward cycle, but they may fail to do so before the start and be still executing a DKG round AFTER the start of a reward cycle.

Ah, yes, you're right. This has to be optional. The signers can't compute the aggregate public key until after DKG, so there's going to be a delay between calculating the reward set and setting the aggregate public key in `pox-4`.

However, I think you'll need to update `get_nakamoto_reward_cycle_info()` to return `None` if the aggregate public key is not known.  We should keep trying to compute the reward set until we have an aggregate public key.

Reply:
> but they may fail to do so before the start and be still executing a DKG round AFTER the start of a reward cycle.

As of right now, if this happens, then the chain will freeze.  There's no way for the Stacks chain state to advance if the next reward cycle begins without the new aggregate public key. 

This could be addressed if there was a Stacks-on-Bitcoin way to set the aggregate public key, however.  I'll open a discussion for it in the SIP repo.

Reply:
https://github.com/stacksgov/sips/discussions/162